provider/redis.go

```diff
@@ -34,58 +38,126 @@ func redisOnlineStoreFactory(serialized pc.SerializedConfig) (Provider, error) {
 	if redisConfig.Prefix == "" {
 		redisConfig.Prefix = "Featureform_table__"
 	}
-	return NewRedisOnlineStore(redisConfig), nil
+	return NewRedisOnlineStore(redisConfig)
 }
 
-func NewRedisOnlineStore(options *pc.RedisConfig) *redisOnlineStore {
-	redisOptions := &redis.Options{
-		Addr:     options.Addr,
-		Password: options.Password,
-		DB:       options.DB,
+func NewRedisOnlineStore(options *pc.RedisConfig) (*redisOnlineStore, error) {
+	redisOptions := rueidis.ClientOption{
+		InitAddress: []string{options.Addr},
+		Password:    options.Password,
+		SelectDB:    options.DB,
+		/*
+			The rueidis client opts-in to server-assisted client-side caching by default.
+			Given we're not making use of this feature (i.e. via the commands `DoCach` or
+			`DoMultiCache`), and given this is normally a feature that a user would have to
+			explicitly opt into for a connection via the Redis command `CLIENT TRACKING ON`,
+			we disable this feature by default to avoid any issues with Redis servers versions
+			under 6.0.
+
+			See details here:
+				* rueidis client-side caching: https://github.com/redis/rueidis#client-side-caching
+				* Redis CLIENT TRACKING command: https://redis.io/commands/client-tracking/
+		*/
```

Comment:
@simba-git and @sdreyer, let me know if this comment doesn't fully explain the decision here. Basically, the `rueidis` library makes a decision to opt users into a configuration that _may_ not be available for all users, and given we're not making use of this feature, I'm disabling it here to return our use of Redis back to its default.